core/arch/arm/mm/core_mmu.c

```diff
@@ -219,10 +219,25 @@ void core_init_mmu_map(void)
 	/* map what needs to be mapped (non-null size and non INTRAM/EXTRAM) */
 	map = in;
 	while (map->type != MEM_AREA_NOTYPE) {
-		if (map->va)
-			panic();
+		if (map->va) {
+#ifdef CFG_WITH_PAGER
+			if (map->va & SMALL_PAGE_MASK)
+				panic();
+
+#else
+			if (map->va & ((1 << CORE_MMU_PGDIR_SHIFT) - 1))
+				panic();
+#endif
+		} else {
+			map->va = map->pa;	/* 1-to-1 pa = va mapping */
+		}
+
+#ifdef CFG_WITH_PAGER
+		map->region_size = SMALL_PAGE_SIZE;
```

Comment:
@jenswi-linaro Hi Jens, is it also valid for MEM_AREA_IO_xx? I add the check to make sure users pass section or page aligned address when they fill the PA/VA address.